java/driver/jdbc/src/main/java/org/apache/arrow/adbc/driver/jdbc/JdbcConnection.java

```diff
@@ -254,6 +254,7 @@ public void setIsolationLevel(IsolationLevel level) throws AdbcException {
   @Override
   public void close() throws Exception {
     connection.close();
+    allocator.close();
```

Comment:
Can we use AutoCloseables here so that an exception raised by one of the close() methods won't affect the other?